src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -147,10 +150,11 @@ private async Task SearchCurrentDocumentAsync(CancellationToken cancellationToke
                 return;
 
             await AddProgressItemsAsync(1, cancellationToken).ConfigureAwait(false);
-            await service.SearchDocumentAsync(
-                _activeDocument, _searchPattern, _kinds, _activeDocument,
-                r => _callback.AddItemAsync(project, r, cancellationToken),
-                cancellationToken).ConfigureAwait(false);
+            await foreach (var item in service.SearchDocumentAsync(
+                _activeDocument, _searchPattern, _kinds, _activeDocument, cancellationToken))
+            {
+                await _callback.AddItemAsync(project, item, cancellationToken).ConfigureAwait(false);
```

Comment:
Should we keep pushing the pattern further up higher, or was this a convenient stopping point?

Reply:
this is effectively the top level entrypoint (or close enough to not make moving up any more matter :))